aspnetcore/fundamentals/logging.md

```diff
@@ -239,33 +262,127 @@ System.Exception: Item not found exception.
+* If no rules are selected, use `MinimumLevel`.
+ 
+For example, suppose you have the preceding list of rules and you create an `ILogger` object for category "Microsoft.AspNetCore.Mvc.Razor.RazorViewEngine":
+
+* For the Debug provider, rules 1, 6, and 8 apply. Rule 8 is most specific, so that's the one selected.
+* For the Console provider, rules 3, 4, 5, and 6 apply. Rule 3 is most specific.
+
+When you create logs with an `ILogger` for category "Microsoft.AspNetCore.Mvc.Razor.RazorViewEngine", logs of `Trace` level and above will go to the Debug provider, and logs of `Debug` level and above will go to the Console provider.
+
+**Provider aliases**
+
+You can use the type name to specify a provider in configuration, but each provider defines a shorter *alias* that is easier to use. For the built-in providers, use the following aliases:
+
+- Console
+- Debug
+- EventLog
+- AzureAppServices
+- TraceSource
+- EventSource
+
+**Default minimum level**
+
+There is a minimum level setting that takes effect only if no rules from configuration or code apply for a given provider and category. The following example shows how to set the minimum level:
+
+[!code-csharp[](logging/sample2/Program.cs?name=snippet_MinLevel&highlight=3)]
+
+IF you don't explicitly set the minimum level, the default value is `Information`, which means that `Trace` and `Debug` logs are ignored.
+
+**Filter functions**
+
+You can write code in a filter function to apply filtering rules. A filter function is invoked for all providers and categories that do not have rules assigned to them by configuration or code. Code in the function has access to the provider type, category, and log level to decide whether or not a message should be logged. For example:
+
+[!code-csharp[](logging/sample2/Program.cs?name=snippet_FilterFunction&highlight=5-13)]
+
+---
+
 ## Log scopes
 
 You can group a set of logical operations within a *scope* in order to attach the same data to each log that is created as part of that set.  For example, you might want every log created as part of processing a transaction to include the transaction ID.
 
 A scope is an `IDisposable` type that is returned by the `ILogger.BeginScope<TState>` method and lasts until it is disposed. You use a scope by wrapping your logger calls in a `using` block, as shown here:
 
-[!code-csharp[](logging/sample/src/TodoApi/Controllers/TodoController.cs?name=snippet_Scopes&highlight=4-5,13)]
+[!code-csharp[](logging/sample//Controllers/TodoController.cs?name=snippet_Scopes&highlight=4-5,13)]
 
 The following code enables scopes for the console provider:
 
-[!code-csharp[](logging/sample/src/TodoApi/Startup.cs?name=snippet_Scopes&highlight=6)]
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+[!code-csharp[](logging/sample//Startup.cs?name=snippet_Scopes&highlight=6)]
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+[!code-csharp[](logging/sample2/Program.cs?name=snippet_Scopes&highlight=4)]
```

Comment:
It isn't immediately obvious which class this code should be included in. For the sake of clarity, I recommend expanding the snippet to include the Program.cs class definition as well.